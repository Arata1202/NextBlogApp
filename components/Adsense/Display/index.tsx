'use client';

import { useEffect } from 'react';
import { usePathname } from 'next/navigation';
import { useTheme } from 'next-themes';

const PUBLISHER_ID = '1705865999592590';

declare global {
  interface Window {
    adsbygoogle: { [key: string]: unknown }[];
  }
}

type DisplayProps = {
  slot: string;
  format?: string;
  responsive?: string;
  style?: any;
};

const Display = ({ slot, format = 'rectangle', responsive = 'false', style }: DisplayProps) => {
  let pathname = usePathname();
  pathname = pathname ? pathname : '';

  useEffect(() => {
    try {
      (window.adsbygoogle = window.adsbygoogle || []).push({});
    } catch (err) {
      console.error(err);
    }
  }, [pathname]);

  const { theme } = useTheme();

  return (
    <div
      style={{ maxWidth: '100%' }}
      className="FirstAd"
      key={pathname.replace(/\//g, '-') + '-' + slot}
    >
      <p className={`text-center ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}>
        スポンサーリンク
      </p>
      <ins
        className="adsbygoogle"
        style={{
          display: 'flex',
          justifyContent: 'center',
          width: '100%',
          height: '250px',
          ...style,
        }}
        data-ad-client={`ca-pub-${PUBLISHER_ID}`}
        data-ad-slot={slot}
        data-ad-format={format}
        data-full-width-responsive={responsive}
      />
    </div>
  );
};

export default Display;
