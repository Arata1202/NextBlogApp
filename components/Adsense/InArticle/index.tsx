'use client';

import { useEffect } from 'react';
import { usePathname } from 'next/navigation';

const PUBLISHER_ID = '1705865999592590';

declare global {
  interface Window {
    adsbygoogle: { [key: string]: unknown }[];
  }
}

type DisplayProps = {
  slot: string;
  format?: string;
  responsive?: string;
  style?: any;
};

const InArticle = ({ slot, format = 'fluid', responsive = 'false', style }: DisplayProps) => {
  let pathname = usePathname();
  pathname = pathname ? pathname : '';

  useEffect(() => {
    try {
      (window.adsbygoogle = window.adsbygoogle || []).push({});
    } catch (err) {
      console.error(err);
    }
  }, [pathname]);

  return (
    <div
      style={{ maxWidth: '100%', overflow: 'hidden' }}
      className="FirstAd"
      key={pathname.replace(/\//g, '-') + '-' + slot}
    >
      <p className="text-center">スポンサーリンク</p>
      <ins
        className="adsbygoogle"
        style={{ display: 'flex', justifyContent: 'center', width: '100%', ...style }}
        data-ad-layout="in-article"
        data-ad-client={`ca-pub-${PUBLISHER_ID}`}
        data-ad-slot={slot}
        data-ad-format={format}
        data-full-width-responsive={responsive}
      />
    </div>
  );
};

export default InArticle;
