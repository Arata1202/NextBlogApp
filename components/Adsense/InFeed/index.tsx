'use client';

import { useEffect } from 'react';
import { usePathname } from 'next/navigation';

const PUBLISHER_ID = '1705865999592590';

declare global {
  interface Window {
    adsbygoogle: { [key: string]: unknown }[];
  }
}

type InFeedProps = {
  slot: string;
  format?: string;
  responsive?: string;
  style?: any;
};

const InFeed = ({ slot, format = 'fluid', responsive = 'true', style }: InFeedProps) => {
  let pathname = usePathname();
  pathname = pathname ? pathname : '';

  useEffect(() => {
    try {
      (window.adsbygoogle = window.adsbygoogle || []).push({});
    } catch (err) {
      console.error(err);
    }
  }, [pathname]);

  return (
    <div key={pathname.replace(/\//g, '-') + '-' + slot}>
      <ins
        className="adsbygoogle"
        style={{ display: 'block', width: '100%', ...style }}
        data-ad-client={`ca-pub-${PUBLISHER_ID}`}
        data-ad-layout-key="-5k+co+1n-b5+e3"
        data-ad-slot={slot}
        data-ad-format={format}
        data-full-width-responsive={responsive}
      />
    </div>
  );
};

export default InFeed;
