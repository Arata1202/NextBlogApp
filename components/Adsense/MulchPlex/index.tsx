'use client';

import { useEffect } from 'react';
import { usePathname } from 'next/navigation';

const PUBLISHER_ID = '1705865999592590';

declare global {
  interface Window {
    adsbygoogle: { [key: string]: unknown }[];
  }
}

type MulchplexProps = {
  slot: string;
  format?: string;
  responsive?: string;
  style?: any;
};

const Mulchplex = ({
  slot,
  format = 'autorelaxed',
  responsive = 'true',
  style,
}: MulchplexProps) => {
  let pathname = usePathname();
  pathname = pathname ? pathname : '';

  useEffect(() => {
    try {
      (window.adsbygoogle = window.adsbygoogle || []).push({});
    } catch (err) {
      console.error(err);
    }
  }, [pathname]);

  return (
    <div
      style={{ maxWidth: '100%', overflow: 'hidden' }}
      className="FirstAd"
      key={pathname.replace(/\//g, '-') + '-' + slot}
    >
      <p className="text-center">スポンサーリンク</p>
      <ins
        className="adsbygoogle"
        style={{ display: 'block', width: '100%', ...style }}
        data-ad-client={`ca-pub-${PUBLISHER_ID}`}
        data-ad-slot={slot}
        data-ad-format={format}
        data-full-width-responsive={responsive}
      />
    </div>
  );
};

export default Mulchplex;
