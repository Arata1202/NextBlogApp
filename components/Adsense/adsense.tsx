import Script from 'next/script';

export const Adsense = () => {
  return (
    <Script
      async
      src={process.env.GOOGLE_ADSENSE_ID}
      crossOrigin="anonymous"
      strategy="afterInteractive"
      data-full-width-responsive="false"
    />
  );
};
