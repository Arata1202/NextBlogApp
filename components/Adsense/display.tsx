'use client';

import { useEffect } from 'react';
import { usePathname } from 'next/navigation';

const PUBLISHER_ID = '1705865999592590';

declare global {
  interface Window {
    adsbygoogle: { [key: string]: unknown }[];
  }
}

type DisplayProps = {
  slot: string;
  format?: string;
  responsive?: string;
  style?: any;
};

const Display = ({ slot, format = 'auto', responsive = 'true', style }: DisplayProps) => {
  let pathname = usePathname();
  pathname = pathname ? pathname : '';

  useEffect(() => {
    try {
      (window.adsbygoogle = window.adsbygoogle || []).push({});
    } catch (err) {
      console.error(err);
    }
  }, [pathname]);

  return (
    <div
      style={{ maxWidth: '100%', overflow: 'hidden' }}
      key={pathname.replace(/\//g, '-') + '-' + slot}
    >
      <p className="text-center">スポンサーリンク</p>
      <ins
        className="adsbygoogle check"
        style={{ display: 'block', width: '100%', ...style }}
        data-ad-client={`ca-pub-${PUBLISHER_ID}`}
        data-ad-slot={slot}
        data-ad-format={format}
        data-full-width-responsive={responsive}
      />
    </div>
  );
};

export default Display;
