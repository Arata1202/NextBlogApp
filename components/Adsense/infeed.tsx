import Script from 'next/script';
export default function InFeed() {
  return (
    <div style={{ maxWidth: '100%' }}>
      <ins
        className="adsbygoogle"
        style={{ display: 'block' }}
        data-ad-format="fluid"
        data-ad-layout-key="-5k+co+1n-b5+e3"
        data-ad-client="ca-pub-1705865999592590"
        data-ad-slot="1678694276"
      ></ins>
      <Script id="adsbygoogle-init" strategy="lazyOnload">
        {`
          (adsbygoogle = window.adsbygoogle || []).push({});
        `}
      </Script>
    </div>
  );
}
