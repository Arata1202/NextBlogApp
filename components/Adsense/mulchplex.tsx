import Script from 'next/script';
export default function Mulchplex() {
  return (
    <div style={{ maxWidth: '100%' }}>
      <ins
        className="adsbygoogle"
        style={{ display: 'block' }}
        data-ad-format="autorelaxed"
        data-ad-client="ca-pub-1705865999592590"
        data-ad-slot="5469260892"
      ></ins>
      <Script id="adsbygoogle-init" strategy="lazyOnload">
        {`
          (adsbygoogle = window.adsbygoogle || []).push({});
        `}
      </Script>
    </div>
  );
}
