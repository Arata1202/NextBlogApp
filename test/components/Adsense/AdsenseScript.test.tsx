import { render } from '@testing-library/react';
import { Adsense } from '@/components/Adsense/AdsenseScript';

describe('AdsenseScript', () => {
  test('AdsenseScript)', () => {
    const { asFragment } = render(<Adsense />);
    expect(asFragment()).toMatchSnapshot();
  });
});
