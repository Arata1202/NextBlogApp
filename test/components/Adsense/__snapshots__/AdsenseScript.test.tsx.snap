// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdsenseScript AdsenseScript) 1`] = `<DocumentFragment />`;
